### PR TITLE
feat(mobile): Add ShareExpenseScreen with split calculator (#218)

### LINKED ISSUE #188: API: GET /api/v1/users/lookup - Search users for expense sharing
## Description
Implement REST API endpoint to search for users by email or name for expense sharing.

## Endpoint
`GET /api/v1/users/lookup`

## Query Parameters
- `q`: string (email or name search)
- `limit`: number (optional, default: 10)

## Response
```json
{
  "success": true,
  "data": {
    "users": [
      {
        "id": "string",
        "name": "string",
        "email": "string (partially masked)"
      }
    ]
  }
}
```

## Acceptance Criteria
- [ ] JWT authentication required
- [ ] Search by email (exact match) or name (partial match)
- [ ] Exclude current user from results
- [ ] Mask email addresses for privacy (show first 2 chars + domain)
- [ ] Limit results to prevent data scraping
- [ ] Rate limit to prevent abuse

### LINKED ISSUE #218: mobile: Create ShareExpenseScreen with split calculator
## Problem
No way to create shared expenses from mobile.

### Screen: ShareExpenseScreen
Modal from SharingScreen FAB

### Form Steps

**Step 1: Expense Details**
- Description (required)
- Total amount (required)
- Date (default today)
- Category picker

**Step 2: Add Participants**
- User lookup by email (GET /api/v1/users/lookup)
- Add multiple participants
- Show participant list

**Step 3: Split Configuration**
- Split type: EQUAL / PERCENTAGE / FIXED
- EQUAL: auto-calculate per person
- PERCENTAGE: input % per person (must sum to 100)
- FIXED: input amount per person (must sum to total)
- Preview split amounts

### Validation
- At least 1 participant
- Split amounts must equal total
- Percentages must sum to 100

### Actions
- Create: calls sharingStore.createSharedExpense
- Cancel: dismisses

## Depends on
- #207 sharingStore
- #182 POST /api/v1/expenses/share
- #188 GET /api/v1/users/lookup

## Acceptance Criteria
- [ ] Multi-step form works
- [ ] User lookup works
- [ ] All split types work
- [ ] Validation prevents invalid splits
- [ ] Creates successfully

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

Implements mobile expense sharing functionality, allowing users to create shared expenses with other users using three split types: EQUAL, PERCENTAGE, and FIXED.

**Key Features:**
- ShareExpenseScreen component with multi-step form flow
- TransactionPickerModal for selecting transactions to share
- Split calculator utility with real-time preview
- User lookup API endpoint with security features
- Comprehensive validation and error handling

**Components Added:**
- `ShareExpenseScreen` - Main share expense form (979 lines)
- `TransactionPickerModal` - Transaction selection modal (332 lines)
- Split calculator utility module
- Share expense validation utilities
- User lookup API endpoint

**Changes:**
- Added `createSharedExpense` action to sharingStore
- Updated navigation types and AppStack
- Wired up SharingScreen "+ Share" button
- Comprehensive test suite (100+ test cases)

## Test Plan

**Unit Tests:**
- ✅ Split calculator tests (68 tests) - all split types, rounding, edge cases
- ✅ Validation tests (93 tests) - XSS prevention, boundary conditions
- ✅ Store action tests (7 tests) - createSharedExpense all split types
- ✅ API endpoint tests (23 tests) - auth, rate limiting, validation

**Manual Testing:**
1. Navigate to SharingScreen
2. Tap "+ Share" button
3. Select a transaction from picker
4. Add participants via email lookup
5. Test all three split types (EQUAL, PERCENTAGE, FIXED)
6. Verify real-time calculation preview
7. Submit and verify shared expense created
8. Verify SharingScreen updates

**Security:**
- Rate limiting on user lookup (5 req/min)
- XSS prevention in description field
- JWT authentication required
- Self-lookup prevention

## Related Issues

Closes #218
Implements GET /api/v1/users/lookup (closes #188)

## Review Notes

- All review feedback addressed (2 iterations)
- AI slop cleaned (184 lines removed)
- Documentation updated (API_AUDIT.md, API_CONTRACTS.md, CHANGELOG.md)
- Test coverage: 90%+ for new code
- TypeScript: No errors
- Build: Passes

### CODE REVIEW

File: mobile/src/lib/splitCalculator.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+      }
+      break;
+    }
+
+    case 'FIXED': {
+      for (const p of participants) {
+        if (p.fixedAmount == null || p.fixedAmount < 0) {
+          errors.push({
+            field: 'amount',
+            message: `Invalid amount for ${p.email}. Must be zero or greater.`,
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply mobile/src/lib/splitCalculator.ts]: ✅ Fixed in commit 00d14cf

Updated validation to enforce minimum share amounts:
- FIXED split: Now requires shareAmount >= 0.01 (was >= 0)
- PERCENTAGE split: Now requires percentage > 0 (was >= 0)

This prevents the client from accepting /usr/bin/bash shares that would be rejected by the API's participantApiSchema validation.
